### PR TITLE
#13141. Avoid conflict in 1to1 calls when two clients answer at same time

### PR DESCRIPTION
**Pull request by alber2510**
If two clients answer an incoming call roughly at same time, a session is established between the 2 clients of the same user (apart from the sessions between the 2 clients and the caller). I could result on the wrong streams being received by the other peer. In example, User A sees video from user B (client 1: B1), but user B in client 2 (B2) sees the video from B1.

**Risk area/s:**
- 1on1 calls answered at two clients at the same time (only one client should success answering the call).
- Group calls answered at two clients of the same user, which is allowed.

### CODE REVIEW

File: src/rtcModule/webrtc.cpp [hunk 1]
```diff
@@ -2359,9 +2392,24 @@ void Call::onClientLeftCall(Id userid, uint32_t clientid)
         {
             mManager.launchCallRetry(mChat.chatId(), sentAv(), false);
         }
+    }
+    else if (mState >= kStateInProgress)
+    {
+        if (userid == mManager.mKarereClient.myHandle())
```
Comment (sergiohs84) [inline src/rtcModule/webrtc.cpp]: Perhaps we need to add `&& !chat.isGroup()` to the `if` above.

File: src/rtcModule/webrtc.cpp [hunk 2]
```diff
@@ -2359,9 +2392,24 @@ void Call::onClientLeftCall(Id userid, uint32_t clientid)
         {
             mManager.launchCallRetry(mChat.chatId(), sentAv(), false);
         }
+    }
```
Comment (sergiohs84) [inline src/rtcModule/webrtc.cpp]: Maybe in the block above we should preserve the `cancelSessionRetryTimer()` and the `destroyIfNoSessionsOrRetries()`?? Not sure, but worths to check it.

File: src/rtcModule/webrtc.cpp (hunk 1 above)
Comment (alber2510) [reply src/rtcModule/webrtc.cpp]: That part of code was added by merge. It's not necessary

File: src/rtcModule/webrtc.cpp (hunk 2 above)
Comment (alber2510) [reply src/rtcModule/webrtc.cpp]: That part of code was added by merge. It's not necessary

**Comment by sergiohs84** [comment]
trigger compilation

**Comment by sergiohs84** [comment]
trigger compilation

**Comment by alber2510** [comment]
trigger compilation

**Pull request merged**